src/mlpack/tests/augmented_rnns_tasks_test.cpp

```diff
@@ -0,0 +1,271 @@
+/**
+ * @file augmented_rnns_tasks.cpp
+ * @author Konstantin Sidorov
+ *
+ * Tests the rtasks for augmented network models.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <vector>
+#include <algorithm>
+#include <utility>
+
+#include <mlpack/core.hpp>
+
+#include <mlpack/core/data/binarize.hpp>
+
+#include <mlpack/methods/ann/augmented/tasks/copy.hpp>
+#include <mlpack/methods/ann/augmented/tasks/sort.hpp>
+#include <mlpack/methods/ann/augmented/tasks/add.hpp>
+#include <mlpack/methods/ann/augmented/tasks/score.hpp>
+
+#include <mlpack/core/optimizers/adam/adam.hpp>
+#include <mlpack/methods/ann/layer/layer.hpp>
+#include <mlpack/methods/ann/layer/leaky_relu.hpp>
+#include <mlpack/methods/ann/rnn.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "test_tools.hpp"
+
+using std::vector;
+using std::pair;
+using std::make_pair;
+
+using namespace mlpack::ann::augmented;
+using namespace mlpack::ann::augmented::tasks;
+using namespace mlpack::ann::augmented::scorers;
+
+using namespace mlpack::ann;
+using namespace mlpack::optimization;
+
+using mlpack::data::Binarize;
+
+// The dummy model that simply copies the sequence
+// the required number of times
+// (yes, no ML here, we're unit testing :)
+class HardCodedCopyModel {
```

Comment:
It would be great if you could go over the test and adjust the style, e.g. put ``{`` on a new line. Also can you comment each function as you did for the rest of the code. I think a lot of people take a look at the test to dive into the codebase and it would be helpful if there are some comments to guide them. Let me know what you think.